core/src/main/scala/cats/syntax/ior.scala

```diff
@@ -0,0 +1,163 @@
+package cats.syntax
+
+import cats.data.{Ior, IorNel, NonEmptyList}
+
+trait IorSyntax {
+  implicit def catsSyntaxIorId[A](a: A): IorIdOps[A] = new IorIdOps(a)
+  implicit def catsSyntaxListIorNel[A, B](list: List[IorNel[A, B]]): IorNelListOps[A, B] =
+    new IorNelListOps(list)
+}
+
+final class IorIdOps[A](val a: A) extends AnyVal {
+  /**
+    * Wrap a value in `Ior.Right`.
+    *
+    * Example:
+    * {{{
+    * scala> import cats.data.Ior
+    * scala> import cats.implicits._
+    *
+    * scala> "hello".rightIor[String]
+    * res0: Ior[String, String] = Right(hello)
+    * }}}
+    */
+  def rightIor[B]: Ior[B, A] = Ior.right(a)
+
+  /**
+    * Wrap a value in `Ior.Left`.
+    *
+    * Example:
+    * {{{
+    * scala> import cats.data.Ior
+    * scala> import cats.implicits._
+    *
+    * scala> "error".leftIor[String]
+    * res0: Ior[String, String] = Left(error)
+    * }}}
+    */
+  def leftIor[B]: Ior[A, B] = Ior.left(a)
+
+  /**
+    * Wrap a value in the right side of `Ior.Both`.
+    *
+    * Example:
+    * {{{
+    * scala> import cats.data.Ior
+    * scala> import cats.implicits._
+    *
+    * scala> "hello".putRightIor("error")
+    * res0: Ior[String, String] = Both(error,hello)
+    * }}}
+    */
+  def putRightIor[B](left: B): Ior[B, A] = Ior.both(left, a)
```

Comment:
I am not sure about the necessity of these syntax: 
`putRightIor` `putLeftIor` `rightIorNel` `leftIorNel` `putRightIorNel` `putLeftIorNel`
I am just not sure if the convenience of these methods overrides the cost of having them on everything.